packages/ember-glimmer/lib/helpers/mut.js

```diff
@@ -0,0 +1,37 @@
+import { assert } from 'ember-metal/debug';
+import { InternalHelperReference, MUT, UPDATE, INVOKE } from '../utils/references';
+import { isConst } from 'glimmer-reference';
+
+export default {
+  isInternalHelper: true,
+  toReference(args) {
+    let ref = args.positional.at(0);
+
+    assert('You can only pass a path to mut', !isConst(ref));
+
+    if (!ref[MUT]) {
+      return new MutHelperReference(mut, args);
+    }
+    return ref;
+  }
+};
+
+function mut({ positional }) {
+  return positional.at(0).value();
+}
+
+class MutHelperReference extends InternalHelperReference {
+   constructor() {
+    super(...arguments);
+    this[MUT] = true;
```

Comment:
I vaguely recall @krisselden mentioning we should use meta for flags like this, maybe he can c/d?